src/System.Security.Cryptography.Csp/tests/PasswordDeriveBytesTests.cs

```diff
@@ -0,0 +1,481 @@
+        {
+            byte[] output;
+
+            using (var deriveBytes = new PasswordDeriveBytes(string.Empty, new byte[8], "SHA1", 1))
+            {
+                output = deriveBytes.GetBytes(1);
+            }
+
+            Assert.Equal(1, output.Length);
+            Assert.Equal(0xF8, output[0]);
+        }
+
+        [Fact]
+        public static void GetBytes_ZeroLength()
+        {
+            using (var deriveBytes = new PasswordDeriveBytes(TestPassword, s_testSalt))
+            {
+                Assert.Throws<ArgumentException>(() => deriveBytes.GetBytes(0));
+            }
+        }
+
+        [Fact]
+        public static void GetBytes_NegativeLength()
+        {
+            using (var deriveBytes = new PasswordDeriveBytes(TestPassword, s_testSalt))
+            {
+                Assert.Throws<OverflowException>(() => deriveBytes.GetBytes(-1));
+                Assert.Throws<OverflowException>(() => deriveBytes.GetBytes(int.MinValue));
+                Assert.Throws<OverflowException>(() => deriveBytes.GetBytes(int.MinValue / 2));
+            }
+        }
+
+        [Fact]
+        public static void GetBytes_StableIfReset()
+        {
+            byte[] first;
+            byte[] second;
+
+            using (var deriveBytes = new PasswordDeriveBytes(TestPassword, s_testSalt))
+            {
+                first = deriveBytes.GetBytes(32);
+                second = deriveBytes.GetBytes(32);
+            }
+
+            Assert.NotEqual(first, second);
+        }
+
+        [Fact]
+        public static void GetBytes_IdempotentIfReset()
+        {
+            byte[] first;
+            byte[] second;
+
+            using (var deriveBytes = new PasswordDeriveBytes(TestPassword, s_testSalt))
+            {
+                first = deriveBytes.GetBytes(32);
+                deriveBytes.Reset();
+                second = deriveBytes.GetBytes(32);
+            }
+
+            Assert.Equal(first, second);
+        }
+
+        [Fact]
+        public static void GetBytes_StreamLike()
+        {
+            byte[] first;
+
+            using (var deriveBytes = new PasswordDeriveBytes(TestPassword, s_testSalt))
+            {
+                first = deriveBytes.GetBytes(32);
+            }
+
+            byte[] second = new byte[first.Length];
+
+            // Reset
+            using (var deriveBytes = new PasswordDeriveBytes(TestPassword, s_testSalt))
+            {
+                byte[] secondFirstHalf = deriveBytes.GetBytes(first.Length / 2);
+                byte[] secondSecondHalf = deriveBytes.GetBytes(first.Length - secondFirstHalf.Length);
+
+                Buffer.BlockCopy(secondFirstHalf, 0, second, 0, secondFirstHalf.Length);
+                Buffer.BlockCopy(secondSecondHalf, 0, second, secondFirstHalf.Length, secondSecondHalf.Length);
+            }
+
+            // Due to "_extraCount" bug in GetBytes() these won't be equal. The bug is fixed in Rfc2898DeriveBytes.
```

Comment:
I did add the happy path and compat tests for this to make this more clear. Yes if you request less than the hash size on the first call to GetBytes() then the "extra" bytes bug comes into play in the next call to GetBytes()